osu.Game/Online/API/APIMod.cs

```diff
@@ -42,7 +42,7 @@ public APIMod(Mod mod)
                 var bindable = (IBindable)property.GetValue(mod);
 
                 if (!bindable.IsDefault)
-                    Settings.Add(property.Name.Underscore(), bindable);
+                    Settings.Add(property.Name.Underscore(), ModUtils.GetSettingUnderlyingValue(bindable));
```

Comment:
One thing which spooks me about this is the "Difficulty Adjust" mod settings (whether it still works just fine after this change). Haven't got around testing that yet though.

Reply:
Works fine as far as I can tell, why do you think it wouldn't?

Reply:
Well with this change, the `Default` value will no longer be copied across to a new mod created by `ToMod`, but upon testing, it seems the refactor of the difficulty adjust mod settings are resilient enough to no longer rely on the `Default` value and `IsDefault` state of the setting. So it seems alright.

Reply:
The check above already excludes default settings from being serialised and existed prior to this change.

Reply:
I was referring to non-default settings with `Default` values populated to them, which used to be the case with difficulty adjust settings prior to the introduction of `DifficultyAdjustSettingControl`s.

Reply:
Shouldn't be any difference. Bindable serialisation [goes via `Value`](https://github.com/ppy/osu-framework/blob/abe410f850c600c98b0787bba4c3a8de16a32c67/osu.Framework/Bindables/Bindable.cs#L404-L407) just as `ModUtils.GetSettingUnderlyingValue` does.

The method I used is even documented to be used for this purpose :smile:  :
https://github.com/ppy/osu/blob/7302e66c5f5706f6e774ece42a5da9d6fadc4667/osu.Game/Utils/ModUtils.cs#L157-L162